src/coreclr/classlibnative/bcltype/objectnative.cpp

```diff
@@ -65,6 +65,34 @@ FCIMPL1(Object*, ObjectNative::GetObjectValue, Object* obj)
 }
 FCIMPLEND
 
+static INT32 TryGetHashCodeHelper(OBJECTREF objRef)
+{
+    DWORD bits = objRef->GetHeader()->GetBits();
+
+    if (bits & BIT_SBLK_IS_HASH_OR_SYNCBLKINDEX)
+    {
+        if (bits & BIT_SBLK_IS_HASHCODE)
+        {
+            // Common case: the object already has a hash code
+            return  bits & MASK_HASHCODE;
+        }
+        else
+        {
+            // We have a sync block index. This means if we already have a hash code,
+            // it is in the sync block, otherwise we generate a new one and store it there
+            SyncBlock *psb = objRef->PassiveGetSyncBlock();
+            if (psb != NULL)
+            {
+                DWORD hashCode = psb->GetHashCode();
+                if (hashCode != 0)
```

Comment:
This branch seems unnecessary given that if it is 0 and we thus avoid returning here, we just fall through to the end where we return 0 anyway.

Reply:
Good point.